project-ivi/ivi
# Implemented visualizer

Closes #28 #44

# Description
This diff connects the navbar to run and step the code from the executor. The information from the interpreter will then sent to the visualizer state manager and rendered in the p5js sketch.

# Change Log
`src/view/components/white_button.js` renamed file for consistency, implemented onClick for component
`src/interpreter/executor/interpret.js`: exported functions for React app to use
`src/view/editor/index.js`: added read-only mode with preformatted code.
`src/view/index.js`: implemented interpreter state. added handler functions for subcomponent behaviors
`src/view/navbar/index.js`: added behavior to Run and Step buttons
`src/view/visualizer/sketch/index.js`: updated to not be a dummy sketch and use the newly introduced sketch state manager
`src/view/visualizer/state.js`: the new sketch state manager that handles incoming interpreter actions

# Manual Test Steps
The Run and Step buttons currently works like this
1. You can start program execution with either Run or Step.
2. Behavior differs between Run and Step here:
    - When you click on Run, the interpreter will automatically step through all steps one at a time at an interval of 1 second.
    - When you click on Step, you will begin the step

Clicking on Run while the interpreter is running will automatically step. Clicking on Run while the interpreter is running automatically will not do anything.

Clicking on Step while the interpreter is running automatically will stop the interpreter from automatically running, and this point on the user will need to step the rest of the way until the program ends.

When the program is running, the editor will be converted to a read-only mode. It will stay in read-only until the program is completed.

<img width="720" alt="screen shot 2017-10-08 at 7 11 17 pm" src="https://user-images.githubusercontent.com/8825042/31321999-8b44276c-ac5c-11e7-9d30-b497aa440cee.png">
<img width="720" alt="screen shot 2017-10-08 at 7 11 27 pm" src="https://user-images.githubusercontent.com/8825042/31321998-8b42b058-ac5c-11e7-92c8-dd5012f6ca9e.png">

You can see the visualizer display the correct variables at each point of the program.
